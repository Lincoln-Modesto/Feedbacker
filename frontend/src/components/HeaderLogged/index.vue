<template>
  <div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
    <div class="w-28 lg:w-36">
      <img class="w-full" src="@/assets/images/logo_white.png" alt="logo">
    </div>

    <div class="flex">
      <ul class="flex list-none">
        <li
        @click="() => router.push({name: 'Credencials'})"
        class="py-2 px-6 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none">
          Credenciais
        </li>
        <li
        @click="() => router.push({name: 'Feedbacks'})"
        class="py-2 px-6 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none">
          Feedbacks
        </li>
        <li
        @click="handleLogout"
        class="py-2 px-6 font-bold bg-white text-brand-main rounded-full cursor-pointer focus:outline-none">
          {{ logoutLabel }}
        </li>
      </ul>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import { useRouter } from 'vue-router'
import { useStore } from '../../hooks/useStore'
import { cleanCurrentUser } from '../../store/user'

export default defineComponent({
  setup () {
    const router = useRouter()
    const store = useStore()

    const logoutLabel = computed(() => {
      if (!store.User.currentUser.name) {
        return '...'
      }
      return `${store.User.currentUser.name} (Sair)`
    })

    function handleLogout () {
      window.localStorage.removeItem('token')
      cleanCurrentUser()
      router.push({ name: 'Home' })
    }

    return {
      router,
      logoutLabel,
      handleLogout
    }
  }
})
</script>
