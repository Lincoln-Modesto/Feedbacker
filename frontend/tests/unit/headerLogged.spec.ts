import { shallowMount } from "@vue/test-utils";
import HeaderLogged from '@/components/HeaderLogged/index.vue'
import { routes } from '@/router'
import { createWebHistory, createRouter } from "vue-router";

const router  = createRouter({
  history: createWebHistory(),
  routes
})
const mockStore = { User: { currentUser: { name: '' }} }

jest.mock('../../src/hooks/useStore', () => {
  return () => {
    return mockStore
  }
})

describe('<HeaderLogged/>', () => {
  it('should render logged correctly', async () => {
    router.push('/')
    await router.isReady()
    mockStore.User.currentUser.name = 'Lincoln'
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [ router ]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it.skip('should render 3 dots when there\'s not user logged', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [ router ]
      }
    })

    const buttonLogout = wrapper.find('#logout-button')
    expect(buttonLogout.text()).toBe('...')
  })

  it('should render username when there\'s user logged', async () => {
    router.push('/')
    await router.isReady()
    mockStore.User.currentUser.name = 'Lincoln'
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [ router ]
      }
    })

    const buttonLogout = wrapper.find('#logout-button')
    expect(buttonLogout.text()).toBe('Lincoln (Sair)')
  })

})
