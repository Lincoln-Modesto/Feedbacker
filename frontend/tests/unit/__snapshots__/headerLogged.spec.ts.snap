// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged/> should render logged correctly 1`] = `
<div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
  <div class="w-28 lg:w-36"><img class="w-full" alt="logo"></div>
  <div class="flex">
    <ul class="flex list-none">
      <li class="py-2 px-6 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Credenciais </li>
      <li class="py-2 px-6 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Feedbacks </li>
      <li id="logout-button" class="py-2 px-6 font-bold bg-white text-brand-main rounded-full cursor-pointer focus:outline-none">Lincoln (Sair)</li>
    </ul>
  </div>
</div>
`;
